### ISSUE TITLE
Add support for cross-compilation to armv7

### ISSUE THREAD

**Issue by mariusor** [mariusor/mpris-scrobbler#25]
Enable meson cross compilation for creating builds for armv7.

Ideally using the Mer environment and scratchbox2 toolkit - this makes it easier to start creating builds using the Mer OBS instance.

**Comment by mariusor** [comment]
Made some progress with this using [Mer Project's OBS](https://build.merproject.org/package/show/home:habarnam:mpris-scrobbler/mpris-scrobbler). Still not finished.

**Comment by Thaodan** [comment]
Maybe add the rpm to the repo (or add another with this as submodule) to build using tar_git in obs?

**Comment by mariusor** [comment]
Hi, thank you for your comment, but unfortunately I'm not very knowledgeable about OBS or creating RPM packages. Do you have some links for me to read in order to understand what you mean?

**Comment by Thaodan** [comment]
You can create a _service file where obs pics up your services.
Tar_git makes a tar out of your repo and makes it available as build source to obs.
You can either point to a stable commit or make rebuild everytime you push something using webhooks.
See a pkg of mine as an example:
https://build.merproject.org/package/view_file/home:Thaodan:testing/communi-sailfish/_service?expand=1

**Comment by Thaodan** [comment]
I'm interresed. I used the maemo-scrobbler before but its based on a Nokia framework and can't be ported.
A accounts plugin to catch oauth token would be great.

**Comment by Thaodan** [comment]
Can you point out whats missing inside the signon plugin?
I thought of creating a plugin that collects the ouath key like the google plugin but just writes it to a text file.
This sounds much better.

**Comment by Thaodan** [comment]
For me librefm is still online?

**Comment by mariusor** [comment]
So, the signon-helper in the current state contains just the services/providers for the 3 scrobbling APIs as used by the Sailfish OS accounts manager. 

If you install it in the SDK they show up in Settings > Accounts. But there's no functionality implemented for any of them.

The authentication steps for libre/last.fm are the same, detailed in the [README](https://github.com/mariusor/mpris-scrobbler#audioscrobbler-compatible).

For listenbrainz it just needs a password input box to put the token from the listenbrainz profile page. Also detailed in the [README](https://github.com/mariusor/mpris-scrobbler#listenbrainz)

To be useful it also needs to implement an ini file writer to path `~/.local/share/mpris-scrobbler/credentials` with the data obtained. 
  
For detailed steps, see the [get_session()](https://github.com/mariusor/mpris-scrobbler/blob/23133415dd1bb8cc93e364a6ceb0d6a09833bcfb/src/signon.c#L85) and [get_token()](https://github.com/mariusor/mpris-scrobbler/blob/23133415dd1bb8cc93e364a6ceb0d6a09833bcfb/src/signon.c#L112) functions in the `signon.c` file.

So basically it's just what you said, only with a pretty Accounts UI. :)

**Comment by Thaodan** [comment]
Could the demon not retrieve the credentials from signond?

**Comment by mariusor** [comment]
I don't know... probably, but I would not want to have the scrobbler too much tied into the sailfish os infrastructure. I'll investigate.
